examples/cluster/plot_cluster_comparison.py

```diff
@@ -35,33 +34,46 @@
 from sklearn.preprocessing import StandardScaler
 from itertools import cycle, islice
 
-np.random.seed(0)
 
-# ============
-# Generate datasets. We choose the size big enough to see the scalability
-# of the algorithms, but not too big to avoid too long running times
-# ============
+# %%
+# Generate sample data
+# ---------------------------------
+
+# We choose the size big enough to see the scalability
+# of the algorithms, but not too big to avoid too long running times.
+# Sample data consists of
+# 1. noisy circles 2. noisy moons 3. blobs with varied variance
+# 4. anisotrophic distributions ('stripes') 5. blobs with smaller variance
+# 6. rectangular noise
+
 n_samples = 500
+# 1. make large circle containing smaller circle
 noisy_circles = datasets.make_circles(n_samples=n_samples, factor=0.5, noise=0.05)
+# 2. make two half circles
 noisy_moons = datasets.make_moons(n_samples=n_samples, noise=0.05)
```

Comment:
I think that we could add a new cell to plot early the datasets used for the experiment:

```python
# %%
fig, axs = plt.subplots(nrows=2, ncols=3, figsize=(9, 6))
for dataset, title, ax in zip(
    [noisy_circles, noisy_moons, varied, aniso, blobs, no_structure],
    ["Concentric circles", "Moons", "Blobs", "Anisotropic blobs", "Blobs", "No structure"],
    axs.flatten(),
):
    X, y = dataset
    ax.scatter(X[:, 0], X[:, 1], marker="o", s=25, edgecolor="k")
    ax.set_xticks(())
    ax.set_yticks(())
    ax.set_title(title)
fig.suptitle("Datasets used for the clustering", fontsize=14)
```

If we add a new figure, then we should think of changing the name of the `png` file to point to the final figure if we are using it in the User Guide. We should search for the file: `auto_examples/cluster/images/sphx_glr_plot_cluster_comparison_001.png` (e.g. it appears in the `clustering.rst` file).